ocaml/xapi-guard/lib/server_interface.ml

```diff
@@ -192,14 +214,22 @@ let serve_forever_lwt_callback_vtpm ~cache mutex vtpm _path _ req body =
 let make_server_varstored ~cache path vm_uuid =
   let module Server =
     Xapi_idl_guard_varstored.Interface.RPC_API (Rpc_lwt.GenServer ()) in
-  let* vm = with_xapi ~cache @@ VM.get_by_uuid ~uuid:vm_uuid in
+  let get_vm_ref () = with_xapi ~cache @@ VM.get_by_uuid ~uuid:vm_uuid in
   let ret v =
     (* TODO: maybe map XAPI exceptions *)
     Lwt.bind v Lwt.return_ok |> Rpc_lwt.T.put
   in
-  let get_nvram _ _ = ret @@ with_xapi ~cache @@ VM.get_NVRAM ~self:vm in
+  let get_nvram _ _ =
+    (let* self = get_vm_ref () in
+     with_xapi ~cache @@ VM.get_NVRAM ~self
+    )
+    |> ret
+  in
   let set_nvram _ _ nvram =
-    ret @@ with_xapi ~cache @@ VM.set_NVRAM_EFI_variables ~self:vm ~value:nvram
+    (let* self = get_vm_ref () in
```

Comment:
we could have a `with_xapi_vm` instead of the separate `get_vm_ref` call, but it is equivalent to the code here.